### PR TITLE
fix `The event objects wrongly wrapped` (close #1277)

### LINKED ISSUE #1277: The event objects wrongly wrapped
Take for example the `message` event:
```javascript
// with hammerhead
JSON.stringify(event); // Uncaught TypeError: Converting circular structure to JSON
Object.keys(event); // ["isTrusted", "data", "origin",  ... and other 26 properties]

// without hammerhead
JSON.stringify(event); // "{"isTrusted":true}"
Object.keys(event); // ["isTrusted"]
```

related with https://github.com/DevExpress/testcafe-hammerhead/issues/1276

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
ping

File: src/client/utils/event-obj-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import fnBind from './fn-bind';
+import nativeMethods from '../sandbox/native-methods';
+
```
Comment (miherlosev) [inline src/client/utils/event-obj-wrapper.js]: Add comment why we do this:
`We cannot redefine 'origin' and 'data' properties using Object.defineProperty method. It does not work in IE and Safari. This is why, we clone event properties and replace some of them.`

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+QUnit.testStart(function () {
+    iframeSandbox.on(iframeSandbox.RUN_TASK_SCRIPT_EVENT, initIframeTestHandler);
+    iframeSandbox.off(iframeSandbox.RUN_TASK_SCRIPT_EVENT, iframeSandbox.iframeReadyToInitHandler);
+});
+
+QUnit.testDone(function () {
+    iframeSandbox.off(iframeSandbox.RUN_TASK_SCRIPT_EVENT, initIframeTestHandler);
+});
+
+function getOwnProperties (obj) {
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: `Object.getOwnPropertyNames`

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+var iframeSandbox = hammerhead.sandbox.iframe;
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: Rename file to the `utils/create-event.js`

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            var iframeWindow = iframe.contentWindow;
+
+            iframeWindow['%hammerhead%'].sandbox.event.message.postMessage(window, ['message', '*']);
+
+            return window.QUnitGlobals.wait(function () {
+                return eventObj && nativeEventObj;
+            });
+        })
+        .then(function () {
+            strictEqual(getOwnProperties(nativeEventObj).sort().join(), getOwnProperties(eventObj).sort().join());
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: split to separte variables, remove `join` and use `deepEqual` for array comparison:
```js
var nativeEventObjOwnProperties = getOwnProperties(nativeEventObj).sort();
var overridenEventObjOwnProperties = getOwnProperties(eventObj).sort();

deepEqual(nativeEventObjOwnProperties, overridenEventObjOwnProperties);
```

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        if (obj.hasOwnProperty(prop))
+            props.push(prop);
+    }
+
+    return props;
+}
+
+module('regression');
+
+test('the "message" event object should be correctly overridden (GH-1277)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: Let's split to two tests:
`test('the overriden "message" event object should have same own properties as native (GH-1277)'

`test('the overriden "message" event object should stringified as native (GH-1277)'

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            iframeWindow['%hammerhead%'].sandbox.event.message.postMessage(window, ['message', '*']);
+
+            return window.QUnitGlobals.wait(function () {
+                return eventObj && nativeEventObj;
+            });
+        })
+        .then(function () {
+            strictEqual(getOwnProperties(nativeEventObj).sort().join(), getOwnProperties(eventObj).sort().join());
+            strictEqual(Object.keys(nativeEventObj).sort().join(), Object.keys(eventObj).sort().join());
+
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: Add comment before `try/catch` block:
`Browser Android 5.1 cannot stringify a native "message" event. It fails with 'Converting circular structure to JSON' error`.

File: test/client/fixtures/utils/event-obj-wrapper-test.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+            try {
+                var nativeEventObjJson = JSON.stringify(nativeEventObj).replace(/"data":\{[^}]*}/, '"data":"message"');
```
Comment (miherlosev) [inline test/client/fixtures/utils/event-obj-wrapper-test.js]: Add comment:
`Browser Safari 9.0 stringify a 'data' property. The overriden event has a wrapped 'data' property. So, we need to trim 'data' to perform comparison.`

**Comment by LavrovArtem** [comment]
FPR

**Pull request merged**